src/query_ctx.h

```diff
@@ -68,6 +87,23 @@ GraphContext *QueryCtx_GetGraphCtx(void);
 /* Retrieve the Redis module context. */
 RedisModuleCtx *QueryCtx_GetRedisModuleCtx(void);
 
+/* Starts a locking flow before commiting changes in the graph and Redis keyspace.
+ * Locking flow is:
+ * 1. LOCK GIL
+ * 2. Key open with `write` flag
+ * 3. Graph R/W lock with write flag
+ * This method returns false if the key has changed from the current graph,
+ * and sets the relevant error message. */
+bool QueryCtx_LockForCommit(void);
+
+/* Starts an ulocking flow and notifies Redis after commiting changes in the graph and Redis keyspace.
+ * Unlocking flow is:
+ * 1. Replicate verbatim
```

Comment:
I thought we were using `replicate` and not `replicateverbatim`

Reply:
replicate it is.